File: queue_array.c
/*
 * C Program to Implement a Queue using an Array
 */
#include <stdio.h>
 
#define MAX 50
int queue_array[MAX];
int rear = - 1;
int front = - 1;
int size=0;
main()
{
    int choice;
    insert(1);
    display();
    
    insert(2);
    display();
    
    insert(3);
    display();
    
    delete();
    display();
    /*
    while (1)
    {
        printf("1.Insert element to queue \n");
        printf("2.Delete element from queue \n");
        printf("3.Display all elements of queue \n");
        printf("4.Quit \n");
        printf("Enter your choice : ");
        scanf("%d", &choice);
        switch (choice)
        {
            case 1:
            insert();
            break;
            case 2:
            delete();
            break;
            case 3:
            display();
            break;
            case 4:
            exit(1);
            default:
            printf("Wrong choice \n");
        }
    }
    */
}
insert(int add_item)
{
    if (rear == MAX - 1)
    printf("Queue Overflow \n");
    else
    {
        if (front == - 1)
        /*If queue is initially empty */
        front = 0;
        rear = rear + 1;
        queue_array[rear] = add_item;
        size++;
    }
} /*End of insert()*/

int isEmpty(){
    if (front == - 1 || front > rear)
    {
        printf("Queue Underflow \n");
        return 1;
    }
    return 0;
}
 
delete()
{
    if( !isEmpty() )
    {
        printf("Element deleted from queue is : %d\n", queue_array[front]);
        front = front + 1;
        size--;
    }
} /*End of delete() */
display()
{
    int i;
    if (front == - 1)
        printf("Queue is empty \n");
    else
    {
        printf("Queue is : \n");
        for (i = front; i <= rear; i++)
            printf("%d ", queue_array[i]);
        printf("\n");
    }
} /*End of display() */

File: Queue_Using_Arrays/Queue_array.java
// Java program to implement a queue using an array 
import java.io.*;
class QueueArr
{
    static int i,front,rear,item,max=5,ch;
    static int a[]=new int[5];
    QueueArr()
    {
        front=-1;
        rear=-1;
    }
    public static void main(String args[])throws IOException
    {
 
        while((boolean)true)
        {
            try
            {
                System.out.println("Select Option 1.insert 2.delete 3.display 4.Exit");
                BufferedReader br=new BufferedReader(new InputStreamReader(System.in));
                ch=Integer.parseInt(br.readLine());
            }
            catch(Exception e)
            {    }
            if(ch==4)
                break;
            else
            {
                switch(ch)
                {
                case 1:
                    insert();
                    break;
                case 2:
                    delete();
                    break;
                case 3:
                    display();
                    break;
                }
            }
        }
    }
    static void insert()
    {
        if(rear>=max)
        {
            System.out.println("Queue is Full");
        }
        else
        {
            try
            {
                BufferedReader br=new BufferedReader(new InputStreamReader(System.in));
                System.out.println("Enter the Element: ");
                item=Integer.parseInt(br.readLine());
            }
            catch(Exception e)
            {}
            rear=rear+1;
            a[rear]=item;
        }
    }
    static void delete()
    {
        if(front==-1)
        {
            System.out.println("Queue is Empty");
        }
        else
        {
            front=front+1;
            item=a[front];
            System.out.println("Deleted Item: "+item);
        }
    }
    static void display()
    {
        System.out.println("Elements in the Queue are:");
        for(int i=front+1; i<=rear; i++)
        {
            System.out.println(a[i]);
        }
    }
}
// sample input and output
//Select Option 1.insert 2.delete 3.display 4.Exit
//1
//Enter the Element:
//12
//Select Option 1.insert 2.delete 3.display 4.Exit
//1
//Enter the Element:
//24
//Select Option 1.insert 2.delete 3.display 4.Exit
//1
//Enter the Element:
//36
//Select Option 1.insert 2.delete 3.display 4.Exit
//3
//Elements in the Queue are:
//12
//24
//36
